### PR TITLE
gitlab CI: remove redundant "dependencies" info

### PR DESCRIPTION
**Pull request by SkySkimmer**
When we started using "needs" for the acyclic graph scheduling we
needed to duplicate dependencies and needs (I forgot if it was
documented or just buggy).

Nowadays gitlab doc https://docs.gitlab.com/ee/ci/yaml/#needs says

>In GitLab 12.6 and later, you can’t combine the dependencies keyword
with needs to control artifact downloads in jobs. dependencies is
still valid in jobs that do not use needs.

Some jobs in stage-1 had "dependencies: []" without a corresponding
"needs", not sure why as the only job in a previous stage is
docker-boot which has no artifacts to skip.

We don't want "needs: []" as that would fail to wait for the docker
image, so we just have nothing instead.

### CODE REVIEW

**Comment by gares** [comment]
Nice! Please wait for the other ci/8.13 related PRs before merging

**Review by Zimmi48 — APPROVED** [review]
Wonderful!

**Comment by Zimmi48** [comment]
> Nice! Please wait for the other ci/8.13 related PRs before merging

OK.

**Comment by Zimmi48** [comment]
BTW, from the link you gave, I discovered the following: https://docs.gitlab.com/ee/ci/yaml/#artifact-downloads-between-pipelines-in-the-same-project

It looks to me like we could save some build time in PR pipelines by reusing build artifacts from the master branch and rebuilding from there instead of rebuilding from zero. It would be particularly useful for PRs changing only a few leaf files in the standard library and pure documentation PRs. WDYT?

**Comment by SkySkimmer** [comment]
Maybe for the dune jobs, I wouldn't trust our makefiles to do the right thing though.

**Comment by SkySkimmer** [comment]
@Zimmi48 @gares can we merge now?

**Comment by Zimmi48** [comment]
Yep! Let's wait for CI to pass though

**Comment by gares** [comment]
yes

**Comment by Zimmi48** [comment]
@coqbot merge now

**Pull request merged**
